src/Microsoft.ComponentDetection.Detectors/pnpm/Pnpm6ComponentDetector.cs

```diff
@@ -0,0 +1,159 @@
+namespace Microsoft.ComponentDetection.Detectors.Pnpm;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.Internal;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.Extensions.Logging;
+
+public class Pnpm6ComponentDetector : FileComponentDetector, IExperimentalDetector
```

Comment:
There is a lot of overlap between this and the v5 detector. What would you say to, in the long run, a `PnpmComponentDetectorFactory` which implemented `FileComponentDetector`, and an `IPnpmDetector` interface. The `PnpmComponentDetectorFactory` should instantiate the correct version of the pnpm detector and pass it the already deserialized lockfile.